lighthouse-core/gather/gatherers/trace-elements.js

```diff
@@ -173,29 +182,42 @@ class TraceElements extends Gatherer {
    * @return {Promise<Array<TraceElementData>>}
    */
   static async getAnimatedElements(passContext, mainThreadEvents) {
-    /** @type Map<number, Set<string>> */
+    /** @type {Map<string, {begin: LH.TraceEvent | undefined, status: LH.TraceEvent | undefined}>} */
+    const animationPairs = new Map();
+    for (const event of mainThreadEvents) {
+      if (event.name !== 'Animation') continue;
+
+      if (!event.id2 || !event.id2.local) continue;
+      const local = event.id2.local;
+
+      const pair = animationPairs.get(local) || {begin: undefined, status: undefined};
+      if (event.ph === 'b') {
+        pair.begin = event;
+      } else if (event.ph === 'n' && event.args.data && event.args.data.compositeFailed) {
+        pair.status = event;
+      }
+      animationPairs.set(local, pair);
+    }
+
+    /** @type Map<number, Set<{animationId: string, failureReasons?: number}>> */
     const elementAnimations = new Map();
-    mainThreadEvents.filter(e => e.name === 'Animation' && e.ph === 'b')
-      .map(e => {
-        return {
-          nodeId: this.getNodeIDFromTraceEvent(e),
-          animationId: this.getAnimationIDFromTraceEvent(e),
-        };
-      })
-      .forEach(({nodeId, animationId}) => {
-        if (!nodeId || !animationId) return;
-        const animationIds = elementAnimations.get(nodeId) || new Set();
-        animationIds.add(animationId);
-        elementAnimations.set(nodeId, animationIds);
-      });
+    for (const [, {begin, status}] of animationPairs) {
+      const nodeId = this.getNodeIDFromTraceEvent(begin);
+      const animationId = this.getAnimationIDFromTraceEvent(begin);
+      const failureReasons = this.getFailureReasonsFromTraceEvent(status);
+      if (!nodeId || !animationId) continue;
+      const animationIds = elementAnimations.get(nodeId) || new Set();
+      animationIds.add({animationId, failureReasons});
+      elementAnimations.set(nodeId, animationIds);
+    }
 
     /** @type Array<TraceElementData> */
     const animatedElementData = [];
     for (const [nodeId, animationIds] of elementAnimations) {
       const animations = [];
-      for (const animationId of animationIds) {
+      for (const {animationId, failureReasons} of animationIds) {
         const animationName = await this.resolveAnimationName(passContext, animationId);
-        animations.push({id: animationId, name: animationName});
+        animations.push({name: animationName, failureReasons});
```

Comment:
can we rename this to `failureReasonsMask` ? descriptive plus differentiates from the `failureReasons` in the audit that's a string.